src/Npgsql/TypeHandlers/RangeHandler.cs

```diff
@@ -9,12 +9,6 @@
 
 namespace Npgsql.TypeHandlers
 {
-    public abstract class RangeHandler : NpgsqlTypeHandler
```

Comment:
Any reason to remove it? Why not to add an abstract method to get all available types?

Reply:
One of the main points of this PR is to remove all the duplicated code in `RangeHandler<TElement>` that was copied as-is from `NpgsqlTypeHandler<T>` (like `Read<TAny>`, `ValidateAndGetLength<TAny>`...). The only reason these couldn't simply be inherited directly from `NpgsqlTypeHandler<T>` (like other type handlers) is that we have the non-generic RangeHandler as a superclass, adding no value at all (I really don't know what I was thinking...).

Reply:
The current trend is to have abstract classes instead interfaces if there is no multiple inheritance. By the way, if I understand Core CLR documentation correctly it says that interface calls are slower than instance calls ([source](https://github.com/dotnet/coreclr/blob/master/Documentation/botr/virtual-stub-dispatch.md#current-state)):

> Currently, VSD is enabled only for interface method calls but not virtual instance method calls. There were several reasons for this:
>
> * **Startup:** Startup working set and speed were hindered because of the need to generate a great deal of initial stubs.
> * **Throughput:** While interface dispatches are generally faster with VSD, virtual instance method calls suffer an unacceptable speed degradation
>
> As a result of disabling VSD for virtual instance method calls, every type has a vtable for virtual instance methods and the implementation table described above is disabled. Dispatch maps are still present to enable interface method dispatching.

Reply:
@YohDeadfall but as I wrote above, the point was to remove the duplication of code from NpgsqlTypeHandler<T> into RangeHandler<T>, which was the case before when RangeHandler<T> inherited from a non-generic RangeHandler.

Regarding perf, this is one of those places where I think we should prefer design correctness and codebase quality over such a micro-optimization. Even if there's a small degradation associated with interface dispatch, this could be a temporary situation and overall I doubt that it'd be decisive.